tests/test_events_api.py

```diff
@@ -0,0 +1,137 @@
+from os import environ
+from typing import Optional
+from unittest.mock import MagicMock, patch
+
+from lbz.aws_boto3 import Boto3Client
+from lbz.events.api import BaseEvent, EventAPI
+
+
+class MyTestEvent(BaseEvent):
+    type = "MY_TEST_EVENT"
+
+
+class TestBaseNewEvent:
+    def test_base_new_event(self) -> None:
+        new_event = MyTestEvent({"x": 1})
+
+        assert hasattr(new_event, "raw_data")
+        assert hasattr(new_event, "data")
+        assert new_event.type == "MY_TEST_EVENT"
+        assert new_event.raw_data == {"x": 1}
+        assert new_event.data == '{"x": 1}'
+
+
+class TestEventApi:
+    event_api: Optional[EventAPI] = None
+
+    def setup_method(self) -> None:
+        # pylint: disable= attribute-defined-outside-init
+        with patch.dict(environ, {"AWS_LAMBDA_FUNCTION_NAME": "million-dollar-lambda"}):
+            self.event_api = EventAPI()
+
+    def teardown_method(self, _test_method) -> None:
+        # pylint: disable= attribute-defined-outside-init
+        self.event_api._del()  # pylint: disable=protected-access
+        self.event_api = None
+
+    def test_init_with_defaults(self) -> None:
+        # pylint: disable=protected-access
```

Comment:
We discussed that multiple times, we shouldn't test protected attributes. They are protected on purpose and shouldn't be used outside the class...

Generally speaking, by these tests, you hardcoded the names of these attributes which is not a good habit.

Reply:
Here you have [the first article I found about this topic](https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html) if you think I am a crazy person touching this topic here 😅